grpc-sys/grpc_wrap.cc

```diff
@@ -845,4 +849,131 @@ grpcwrap_ssl_server_credentials_create(
   return creds;
 }
 
+using grpc::CompletionQueue;
+
+/** Wrapper of grpc::CompletionQueue.
+ */
+typedef struct grpcwrap_completion_queue {
+  CompletionQueue* cq;
+} grpcwrap_completion_queue;
+
+/** Create a ComletionQueue, a shadow of grpc_completion_queue.
+ */
+GPR_EXPORT grpcwrap_completion_queue *GPR_CALLTYPE
+grpcwrap_completion_queue_shadow(grpc_completion_queue *cq) {
+  void *ptr = gpr_malloc(sizeof(CompletionQueue));
+  CompletionQueue *shadow_cq = new(ptr) CompletionQueue(cq);
+
+  grpcwrap_completion_queue *cq_wrapper = (grpcwrap_completion_queue *)
+    gpr_malloc(sizeof(grpcwrap_completion_queue));
+  cq_wrapper->cq = shadow_cq;
+
+  return cq_wrapper;
+}
+
+using grpc::Alarm;
+
+/** Wrapper of grpc::Alarm.
+ */
+typedef struct grpcwrap_alarm {
+  Alarm* alarm;
+} grpcwrap_alarm;
+
+/** Create a completion queue alarm instance */
+GPR_EXPORT grpcwrap_alarm *GPR_CALLTYPE
+grpcwrap_alarm_create(void) {
+  void *ptr = gpr_malloc(sizeof(Alarm));
```

Comment:
em, seem you can malloc with `sizeof(Alarm) + sizeof(grpcwrap_alarm)` for both Alarm and wapper, so we don't need to alloc and dealloc twice.